dali/imgcodec/decoders/decoder_test_helper.h

```diff
@@ -0,0 +1,230 @@
+  */
+  Tensor<CPUBackend> ReadReferenceFrom(const std::string &reference_path) {
+    auto src = FileStream::Open(reference_path, false, false);
+    return ReadReference(src.get());
+  }
+
+  /**
+  * @brief Reads the reference image from specified stream.
+  */
+  virtual Tensor<CPUBackend> ReadReference(InputStream *src) = 0;
+
+ protected:
+  /**
+  * @brief Creates a decoder instance, working on a specified thread pool.
+  */
+  virtual std::shared_ptr<ImageDecoderInstance> CreateDecoder(ThreadPool &tp) = 0;
+
+  /**
+  * @brief Creates a parser to be used.
+  */
+  virtual std::shared_ptr<ImageParser> CreateParser() = 0;
+
+ private:
+  std::string PrettyPosition(size_t index, TensorShape<> shape) const {
+    std::vector<int> pos(shape.size());
+    for (size_t i = pos.size() - 1; i > 0; i--) {
+      pos[i] = index % shape[i];
+      index /= shape[i];
+    }
+
+    std::ostringstream result;
+    result << "(";
+    for (size_t i = 0; i < pos.size(); i++) {
+      result << pos[i];
+      if (i < pos.size() - 1)
+        result << ", ";
+      else
+        result << ")";
+    }
+    return result.str();
+  }
+
+  std::shared_ptr<ImageDecoderInstance> decoder_ = nullptr;
+  std::shared_ptr<ImageParser> parser_ = nullptr;
+  ThreadPool tp_;
+};
+
+
+/**
+* @brief Base class template for tests comparing decoder's results with numpy files.
+*
+* @tparam OutputType Type, to which the image should be decoded.
+*/
+template<typename OutputType>
+class NumpyDecoderTestBase : public CpuDecoderTestBase<OutputType> {
+ private:
+  /**
+  * @brief Converts a tensor to OutputType.
+  */
+  template<typename InputType>
+  Tensor<CPUBackend> Convert(const Tensor<CPUBackend> &in) {
+    Tensor<CPUBackend> out;
+    out.Resize(in.shape(), type2id<OutputType>::value);
+    auto in_view = view<const InputType>(in), out_view = view<OutputType>(out);
+    for (int i = 0; i < volume(in.shape()); i++) {
+      out_view.data[i] = ConvertSatNorm<OutputType>(in_view.data[i]);
+    }
+    return out;
+  }
+
+  /**
+  * @brief Reads a tensor from numpy file.
+  */
+  Tensor<CPUBackend> ReadNumpy(InputStream *src) {
+    numpy::HeaderData header;
+    numpy::ParseHeader(header, src);
+    src->SeekRead(header.data_offset, SEEK_SET);
+
+    Tensor<CPUBackend> data;
+    data.Resize(header.shape, header.type());
+    Index ret = src->Read(static_cast<uint8_t*>(data.raw_mutable_data()), header.nbytes());
+    EXPECT_EQ(ret, (Index)header.nbytes()) << "Failed to read reference numpy file";
+
+    if (header.fortran_order) {
+      Tensor<CPUBackend> transposed;
+      transposed.Resize(data.shape(), data.type());
+      SampleView<CPUBackend> input(data.raw_mutable_data(), data.shape(), data.type());
+      SampleView<CPUBackend> output(transposed.raw_mutable_data(), transposed.shape(),
+                                    transposed.type());
+      numpy::FromFortranOrder(output, input);
+      return transposed;
+    }
+    return data;
+  }
```

Comment:
Move this to `numpy` header or generic test headers? In any case, this should net be a member function of any class.

Reply:
Yeah, probably `numpy.h` would be a better place for that